modules/standard/Path.chpl

```diff
@@ -182,25 +183,27 @@ proc commonPath(in paths: string ...?n): string {
     return firstPath;
   }
 
-  var prefixArray = firstPath.split(pathSep, -1, false);
-  // array of resultant prefix string
+  var prefixList = new list(string);
+  for x in firstPath.split(pathSep, -1, false) do
+    prefixList.append(x);
 
-  var pos = prefixArray.size;   // rightmost index of common prefix
-  var minPathLength = prefixArray.size;
+  var pos = prefixList.size;   // rightmost index of common prefix
+  var minPathLength = prefixList.size;
```

Comment:
What's the motivation for rewriting these idioms that use split() to create an array?  I'm not seeing calls to the newly-deprecated routines, but could easily be missing something.

Reply:
I believe that is because later on the function may make calls to `array.remove`, IE:

```
211   if (flag == 1) {
212     prefixArray.remove(pos..prefixArray.size);
213   } else {
214     prefixArray.remove(minPathLength+1..prefixArray.size);
215     // in case all paths are subsets of the longest path thus pos was never
216     // updated
217   }
```

Reply:
I wonder, should we consider having a version of `split` that returns a list (maybe by taking a param argument to flip between the return types or something)?  Or making `split` always return a list if we think that is the more common use case?

Reply:
Hm, then that wouldn't be specific to `split`, so I thought. However, this `split` here is an iterator, so perhaps the ideal way to code up this is to pass the iterator straight to a list initializer:
```chpl
var prefixList = new list(firstPath.split(pathSep, -1, false));
```

For that, the initializer would take an _iteratorRecord and iterate over it, like proc chpl__initCopy(ir: _iteratorRecord) at ChapelArray.chpl:4357:
```chpl
proc init(iterable: _iteratorRecord) {
  elemType = iteratorToArrayElementType(iterable.type);
  this.complete();
  for elm in iterable do
    this.append(elm);
}
```

Reply:
Separately, switching from array to `list` is not a good enough reason to rename a variable. In most cases, a variable's name has no business reflecting its type. So, `prefixArray` would be a perfectly fine name even if its type is `list`.

Reply:
> Separately, switching from array to list is not a good enough reason to rename a variable.

I disagree.  Maybe the type shouldn't be in the variable in the first place, but if there is a type in the name it *definitely* shouldn't be the wrong type.

Reply:
I agree with Lydia and was logging in to say similarly.  If `myInt` ceases to have integer nature, we shouldn't leave the name the same to minimize diffs.  The current state of the code is ultimately more important than minimizing the diffs for a given PR.

Reply:
OTOH we do not have to interpret the name `prefixArray` to mean literally "its type is a Chapel array". We can interpret this name as "this is a collection that looks like an array". For which `list` is a perfectly legitimate implementation.

Regardless, I am OK with changing the name to `prefixList`.